cli/main.rs

```diff
@@ -361,19 +360,18 @@ async fn create_graph_and_maybe_check(
   );
   let maybe_locker = lockfile::as_maybe_locker(ps.lockfile.clone());
   let maybe_imports = ps.options.to_maybe_imports()?;
-  let maybe_import_map_resolver =
-    ps.maybe_import_map.clone().map(ImportMapResolver::new);
-  let maybe_jsx_resolver = ps
-    .options
-    .to_maybe_jsx_import_source_config()
-    .map(|cfg| JsxResolver::new(cfg, maybe_import_map_resolver.clone()));
-  let maybe_resolver = if maybe_jsx_resolver.is_some() {
-    maybe_jsx_resolver.as_ref().map(|jr| jr.as_resolver())
-  } else {
-    maybe_import_map_resolver
-      .as_ref()
-      .map(|im| im.as_resolver())
-  };
+  let maybe_jsx_config = ps.options.to_maybe_jsx_import_source_config();
+  let maybe_cli_resolver =
+    if maybe_jsx_config.is_some() || ps.maybe_import_map.is_some() {
+      Some(CliResolver::new(
+        maybe_jsx_config,
+        ps.maybe_import_map.clone(),
+      ))
+    } else {
+      None
+    };
```

Comment:
Ideally I wouldn't have to do this conditional and always create `CliResolver` with both options being `None`. However it failed tests - I'd like to revisit it later and clean up further, but current change unblocks me on the immediate need for `npm` specifier support.

Reply:
Can we make a `CliResolver::maybe_new(...) -> Option<Self>`? That would help bury a bunch of these conditions.

Reply:
Good idea 👍